File: src/main/java/com/github/alexandrecarlton/idea/settings/layout/editor/general/auto_import/JavaAutoImportSettings.java
package com.github.alexandrecarlton.idea.settings.layout.editor.general.auto_import;

import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import org.immutables.value.Value;

import java.util.List;

@Value.Immutable
@JsonDeserialize(as = ImmutableJavaAutoImportSettings.class)
public interface JavaAutoImportSettings {

  // This saves to the workspace file, which we don't know how to persist (yet).
  // Optional<Boolean> optimizeImportsOnTheFly();

  List<String> excludeFromImportAndCompletion();

}

File: src/main/kotlin/com/github/alexandrecarlton/idea/settings/layout/editor/general/auto_import/JavaAutoImportSettings.kt
package com.github.alexandrecarlton.idea.settings.layout.editor.general.auto_import

import com.fasterxml.jackson.annotation.JsonProperty

data class JavaAutoImportSettings(

    @JsonProperty("Optimize imports on the fly (for current project)")
    val optimizeImportsOnTheFly: Boolean? = null,

    @JsonProperty("Exclude from import and completion")
    val excludeFromImportAndCompletion: List<String>? = null
)
